src/target/imxrt.c

```diff
@@ -463,8 +463,8 @@ static void imxrt_spi_wait_complete(target_s *const target)
 #ifdef ENABLE_DEBUG
 		/* Read out the status code and display it */
 		const uint32_t status = target_mem_read32(target, IMXRT_FLEXSPI1_STAT1);
-		DEBUG_TARGET(
-			"Error executing sequence, offset %u, error code %u\n", (status >> 16U) & 0xfU, (status >> 24U) & 0xfU);
+		DEBUG_TARGET("Error executing sequence, offset %" PRIu8 ", error code %" PRIu8 "\n",
+			(unsigned char)((status >> 16U) & 0xfU), (unsigned char)((status >> 24U) & 0xfU));
```

Comment:
Please us `uint8_t` here, not the raw C type. This is great otherwise. Technically, you don't need to enclose the expression in brackets because of that final masking operation, but we understand why you've done it this way. `(uint8_t)(status >> 16U) & 0xfU` will execute identically though.

```suggestion
		DEBUG_TARGET("Error executing sequence, offset %u, error code %u\n",
			(uint8_t)(status >> 16U) & 0xfU, (uint8_t)(status >> 24U) & 0xfU);
```

Reply:
Alright, I've made that change, and removed the extra parenthesis.